invenio_drafts_resources/services/records/components/media_files.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020-2021 CERN.
+# Copyright (C) 2021 Northwestern University.
+# Copyright (C) 2023 Graz University of Technology.
+#
+# Invenio-Drafts-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Records service component base classes."""
+from invenio_i18n import gettext as _
+from invenio_records_resources.services.records.components import (
+    MediaFilesOptionsComponent,
+)
+from marshmallow import ValidationError
+
+from .files import DraftFilesComponent
+
+
+class DraftMediaFilesComponent(DraftFilesComponent):
+    """Draft media files component."""
+
+    _files_attr_key = "media_files"
+    _files_data_key = "media_files"
+    _files_bucket_attr_key = "media_bucket"
+    _files_bucket_id_attr_key = "media_bucket_id"
+
+    def __init__(self, service, *args, **kwargs):
+        """Constructor."""
+        super().__init__(service)
+        self.files_component = MediaFilesOptionsComponent(service)
```

Comment:
I'm not sure I'm understanding the inheritance vs composition here.

In records-resources:

1. [The `BaseFilesOptionsComponent` inherits `ServiceComponent`](https://github.com/inveniosoftware/invenio-records-resources/pull/475/files#diff-c8b3bb7ffe826d4540bf3b1a8c798f2651b5de007ddb43aa367aae13ee5d9e96R20). It defines some attributes and methods and overwrites some methods (update and create)
2. In this module `DraftFilesComponent` inherits `ServiceComponent, BaseRecordFilesComponent`
    - The second (`BaseFilesOptionsComponent`) already inherits `ServiceComponent` so the first addition is redundant
    - `DraftFilesComponent` is redefining all the base `_` attributes to assign them a value. This could be solved by inheriting `FilesOptionsComponent`. However, this would be *wrong* (thus, invalidating my suggestion above), because the  `FilesOptionsComponent` (or children classes) are used as `files_components`.
    - `self.files_component` is only used to call the new defined methods (I do not see any call to the overwritten `create` and `update`.
    
TL;DR: There is a big tangling between composition (i.e. `files_component = FilesOptionsComponent(...)`) and inheritance (i.e. assigning value to `_files_attr_key` and company).

Suggestion: Make `FilesOptionsComponent` (and parent/child class) a mixin. That way, we can inherit and access both attributes and new methods. Without the need to change the consturctor signature.